File: backend/src/main/java/com/form/org/Services/EmployeService.java
package com.form.org.Services;

import java.util.List;

import com.form.org.dto.ChangerMotDePasseUtilisateurDTO;
import com.form.org.dto.EmployeDTO;

public interface EmployeService {
	
EmployeDTO save(EmployeDTO dto);
		
EmployeDTO findById(Integer idEmploye);
		
	 List<EmployeDTO>findAll(); 
		
	 void delete(Integer idEmploye);
	 
	 EmployeDTO findByEmail(String email);
	 
	 EmployeDTO changerMotDePasse(ChangerMotDePasseUtilisateurDTO dto); 

}

File: FrontEnd1/src/app/services/employe/employe.service.ts
import { Injectable } from '@angular/core';
import {PatientDTO} from '../../../gs-api/src/models/patient-dto';
import {Observable, of} from 'rxjs';
import {TestCovidv1employeService} from '../../../gs-api/src/services/test-covidv-1employe.service';
import {EmployeDTO} from '../../../gs-api/src/models/employe-dto';

@Injectable({
  providedIn: 'root'
})
export class EmployeService {

  constructor(private  testCovidv1employeService: TestCovidv1employeService) { }

  enregistrerEmploye(employeDt: EmployeDTO): Observable<PatientDTO> {
    return this.testCovidv1employeService.save(employeDt);
  }
  findAllEmploye(): Observable<EmployeDTO[]> {
    return this.testCovidv1employeService.findAll();
  }

  findEmployetById(idEmploye?: number): Observable<EmployeDTO> {
    if (idEmploye) {
      return this.testCovidv1employeService.findById(idEmploye);
    }
    return of();
  }
  findCountEmployees(): Observable<number>{
    return this.testCovidv1employeService.findCountEmploye();
  }
  deleteemploye(idEmploye?: number): Observable<any>{
    if (idEmploye) {
      return this.testCovidv1employeService.delete(idEmploye);
    }
    return of();
  }
}
